File: frontend/src/app/rides/service/ride-estimate.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { RideEstimateRequestDTO, RideEstimateResponseDTO } from '../model/ride-estimate.model';

@Injectable({ providedIn: 'root' })
export class RideEstimateService {

  private readonly baseUrl = 'http://localhost:8080'; 

  constructor(private http: HttpClient) {}

  estimateRide(payload: RideEstimateRequestDTO): Observable<RideEstimateResponseDTO> {
    return this.http.post<RideEstimateResponseDTO>(
      `${this.baseUrl}/api/rides/estimate`,
      payload
    );
  }
}

File: backend/src/main/java/Nuvola/Projekatsiit2025/services/RideEstimateService.java
package Nuvola.Projekatsiit2025.services;

import Nuvola.Projekatsiit2025.dto.RideEstimateRequestDTO;
import Nuvola.Projekatsiit2025.dto.RideEstimateResponseDTO;

public interface RideEstimateService {
    RideEstimateResponseDTO estimateRide(RideEstimateRequestDTO request);
}
